test/functional/p2p_invalid_messages.py

```diff
@@ -42,13 +43,30 @@ def set_test_params(self):
         self.setup_clean_chain = True
 
     def run_test(self):
+        self.test_buffer()
         self.test_magic_bytes()
         self.test_checksum()
         self.test_size()
         self.test_msgtype()
         self.test_large_inv()
         self.test_resource_exhaustion()
 
+    def test_buffer(self):
+        self.log.info("Test message with header split across two buffers")
```

Comment:
suggestion: mention the expected result in the log here, e.g. succeeds, logs x, is sent successfully, etc.

Reply:
We don't currently log that an incomplete header was received.  IMO we shouldn't, it's just an implementation detail of sockets, it doesn't really mean anything on any Bitcoin level.  At the very least maybe changing bitcoind logging is out of scope for this PR?

Reply:
The idea is to mention the expected result. Ther rest is an example, success, disconnects, etc.

Reply:
> We don't currently log that an incomplete header was received.

Yet that suggestion works?

Reply:
In #19302 you added `LogPrintf("Incomplete header");`

Reply:
@jonatack are you sure? I don't see anywhere in the product code that logs "Incomplete headerReceived" (and would be surprised to, since this is purely an implementation detail)

Reply:
> 
> 
> The idea is to mention the expected result. The rest was an example, success, failure, disconnects peer, etc.

Ah I agree, I'll make this a bit more descriptive

Reply:
Oh right, just sockets. Agreed.

Reply:
> @jonatack are you sure? I don't see anywhere in the product code that logs "Incomplete headerReceived" (and would be surprised to, since this is purely an implementation detail)

Weird. I think it was a non-consistent result. Sorry about that.

Reply:
> In [#19302 (comment)](https://github.com/bitcoin/bitcoin/pull/19302#discussion_r441521857) you added `LogPrintf("Incomplete header");`

Hm, it's possible I didn't rebuild properly. It's late here :) will finish review of your net headers refactoring tomorrow fresh and early